Zend/zend_alloc.c

```diff
@@ -2015,7 +2015,7 @@ ZEND_API size_t zend_mm_gc(zend_mm_heap *heap)
 				i++;
 			}
 		}
-		if (chunk->free_pages == ZEND_MM_PAGES - ZEND_MM_FIRST_PAGE) {
+		if (chunk->free_pages == ZEND_MM_PAGES - ZEND_MM_FIRST_PAGE && chunk != heap->main_chunk) {
```

Comment:
I have not encountered this problem. But I think it's necessary to add this check, although it's not  easy to trigger.

Strongly Agree:
> I don't see anything obviously guaranteeing that at least will chunk will stay in main_chunk, and if there was it might accidentally break in a refactoring.

And I am going to make a new pull request its target branch will be PHP-8.1. Thanks for reminding!

Reply:
Unrelated: If you created a branch on your repo rather than using `master`, you would be able to rebase that branch against 8.1 and force push (and change the target branch to 8.1 in the github ui) (and it wouldn't interfere with other branches if you developed those concurrently)

Not actionable, anyway

Reply:
It's so kind of you!

I didn't even know the function before:
> and change the target branch to 8.1 in the github ui

Then after search: [https://github.blog/2016-08-15-change-the-base-branch-of-a-pull-request/](2016-08-15-change-the-base-branch-of-a-pull-request)

Sorry for my knowless. For further discussion please go to: #9865